iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1154,7 +1154,15 @@ private async Task<Message> SendTwinRequestAsync(Message request, string rid, Ca
                         {
                             if (status >= 300)
                             {
-                                throw new IotHubException($"Request {rid} returned status {status}", isTransient: false);
+                                // Retry for Http status code 429 (too many requests)
+                                if (status == 429)
```

Comment:
Did we decide to not retry for 5xx error codes or is service not sending those error codes back? It might be worth putting a comment in calling out why 429 is the only error code we're retrying for.

Reply:
The Hub team is still refactoring the retriable status codes and they will loop us in once they have a proposal. For now there is no breaking change for the existing ones (like 429 here). I can check with Hub again for their refactoring next week, but it appears to be expected that we may bring more retriable codes here in the future.